src/gt4py/next/ffront/decorator.py

```diff
@@ -351,6 +359,44 @@ def _column_axis(self):
         return iter(scanops_per_axis.keys()).__next__()
 
 
+@dataclasses.dataclass(frozen=True)
+class ProgramWithBoundArgs(Program):
+    bound_args: dict[str, typing.Union[float, int, bool]] = None
+
+    def _process_args(self, args: tuple, kwargs: dict):
+        args = list(args)
+        b_args = self.bound_args
+        if (len(args) + len(b_args)) != len(self.past_node.params):
```

Comment:
As discussed this checks do not cover all possible cases of the user giving you wrong arguments. Please extend this and the tests to cover them.